### PR TITLE
APPS-1869 upload log on nextflow head job failure

### PR DESCRIPTION
**Pull request by YuxinShi0423**

### CODE REVIEW

**Comment by YuxinShi0423** [comment]
https://jenkins-vstg.internal.dnanexus.com/job/dx-toolkit/job/dxpy-branch-integration-tests-on-staging-20.04-python3/509/

File: src/python/dxpy/templating/templates/nextflow/src/nextflow.sh [hunk 1]
```diff
@@ -132,20 +132,26 @@ on_exit() {
...
-    mv "$LOG_NAME" "/home/dnanexus/out/nextflow_log/$LOG_NAME" || true
+    echo "=== Execution complete — upload nextflow log to job output destination ${DX_JOB_OUTDIR%/}"
+    NEXFLOW_LOG_ID=$(dx upload "/home/dnanexus/out/nextflow_log/$LOG_NAME" --path "${DX_JOB_OUTDIR%/}/${LOG_NAME}" --wait --brief --no-progress --parents) &&
+      echo "Upload nextflow log as file: $NEXFLOW_LOG_ID" ||
+      echo "Failed to upload log file of current session $NXF_UUID"
   else
-    echo "No nextflow log file available."
+    echo "=== Execution complete — no nextflow log file available."
   fi
+  rm $LOG_NAME || true
```
Comment (mhrvol) [inline src/python/dxpy/templating/templates/nextflow/src/nextflow.sh:143]: It probably does not matter as it is uploaded to the platform, but why do we remove the log here? Isn't it redundant?

**Review by mhrvol — APPROVED** [review]
LGTM

**Comment by YuxinShi0423** [comment]
https://jenkins-vstg.internal.dnanexus.com/job/dx-toolkit/job/dxpy-master-integration-tests-on-staging-16.04-python2/2305/

**Pull request merged**